scientific_library/tvb/simulator/simulator.py

```diff
@@ -363,6 +399,46 @@ def _loop_monitor_output(self, step, state):
         if any(outputi is not None for outputi in output):
             return output
 
+    def bound_and_clamp(self, state):
+        # If there is a state boundary...
+        if self.integrator.state_variable_boundaries is not None:
+            # ...use the integrator's bound_state
+            self.integrator.bound_state(state)
+        # If there is a state clamping...
+        if self.integrator.clamped_state_variable_values is not None:
+            # ...use the integrator's clamp_state
+            self.integrator.clamp_state(state)
+
+    def _print_progression_message(self, step, n_steps):
```

Comment:
I already asked for this to be refactored the progress monitor.  If you want to hardcode a default monitor in the cosimulator, that's fine, but no sys.stdout here please.